### ISSUE TITLE
"Align declarations": Line break if the line gets too long

### ISSUE THREAD

**Issue by ConjuringCoffee** [SAP/abap-cleaner#77]
Hi Jörg-Michael, the rule "Align declarations" doesn't seem to have any logic to add or keep line breaks if the line gets too long.

Example:

```abap
INTERFACE lif_example_interface.
  TYPES this_is_a_type_name TYPE char1.

  CONSTANTS:
    BEGIN OF values_for_the_type,
      first_value TYPE this_is_a_type_name VALUE 'A',
    END OF values_for_the_type.

ENDINTERFACE.


CLASS lcl_example DEFINITION CREATE PRIVATE.

  PUBLIC SECTION.

  PROTECTED SECTION.

  PRIVATE SECTION.
    DATA this_is_an_attribute TYPE lif_example_interface=>this_is_a_type_name VALUE lif_example_interface=>values_for_the_type-first_value.
ENDCLASS.


CLASS lcl_example IMPLEMENTATION.

ENDCLASS.
```

In this case I'd like to be able to put `VALUE lif_example_interface=>values_for_the_type-first_value` on a new line. The ABAP Cleaner rule doesn't allow this.

Here are the options I am using:

![image](https://github.com/SAP/abap-cleaner/assets/72548231/1786fd7d-b412-43e8-8245-352f40ead3ba)

**Comment by jmgrassau** [comment]
Hi ConjuringCoffee, 

I haven't encountered such an example yet, but of course it makes sense to break the line here. So, I think an option for maximum line length should be added. However, where exactly would you expect ```VALUE``` to be moved? Maybe below ```TYPE```? Or below the variable name? 

Kind regards,
Jörg-Michael

**Comment by ConjuringCoffee** [comment]
I think the following two approaches would look good:

```abap
    DATA this_is_an_attribute TYPE  lif_example_interface=>this_is_a_type_name 
                              VALUE lif_example_interface=>values_for_the_type-first_value.
```

```abap
    DATA this_is_an_attribute TYPE lif_example_interface=>this_is_a_type_name 
                              VALUE lif_example_interface=>values_for_the_type-first_value.
```

**Comment by jmgrassau** [comment]
Hi ConjuringCoffee, 

thanks for this issue! I now added a new option to the "Align declarations" rule:

![image](https://github.com/SAP/abap-cleaner/assets/79639298/9063f6d8-5577-4e15-a15c-1edcd6cc82a6)

This changes 

```ABAP
    DATA this_is_an_attribute TYPE lif_example_interface=>this_is_a_type_name VALUE lif_example_interface=>values_for_the_type-first_value.
```

into

```ABAP
    DATA this_is_an_attribute TYPE lif_example_interface=>this_is_a_type_name
                              VALUE lif_example_interface=>values_for_the_type-first_value.
```

or – if you'd only allow a maximum line length of 89 or less – even into

```ABAP
    DATA this_is_an_attribute TYPE lif_example_interface=>this_is_a_type_name
         VALUE lif_example_interface=>values_for_the_type-first_value.
```

This also works in the other direction, moving VALUE clauses behind the type if maximum line length allows for it.

Kind regards,
Jörg-Michael

**Comment by jmgrassau** [comment]
Hi ConjuringCoffee, 

the new option is now available with version 1.5.1 which was just released!

Kind regards,
Jörg-Michael
